src/lib/fountain/FountainDecoder.h

```diff
@@ -46,21 +46,31 @@ class FountainDecoder
 		return _res;
 	}
 
-	std::optional<std::vector<uint8_t>> decode(unsigned block_num, uint8_t* data, size_t length)
+	bool decode(unsigned block_num, uint8_t* data, size_t length)
 	{
 		auto pear = _seenBlocks.insert(block_num);
 		if (!pear.second)
-			return std::nullopt;
+			return false;
```

Comment:
My ideal api here is that we'd call `write()`, and trigger write->recover->decompress->disk write.

For the first pass, I didn't want to hide all the individual steps. I might put it back later and clean up the api.

With the added knowledge of the zstd_decompressor inBuff situation (see whining above), I think having the caller own the wirehair_recover buffer might be a mistake. And if we need to own the memory here anyway, that swings the ideal implementation (as I judge it) back to how it was before these changes. So.. stay tuned